site/documentation/content/api/command-router/index.md

```diff
@@ -0,0 +1,174 @@
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Unregister command consumer for device
+
+Clients use this command to *unregister* a command consumer for the given device. The consumer registration entry to be removed is identified by the provided protocol adapter instance handling command & control messages for a given device. The registration entry for the device is only removed, if the adapter instance identifier of the current registration entry matches the one given in the request.
+
+This API doesn't mandate checks on the validity of the given device in order not to introduce a dependency on the *Device Registration API*. However, implementations of this API may choose to perform such checks or impose a restriction on the overall amount of data that can be stored per tenant in order to protect against malicious requests.
+
+**Message Flow**
+
+
+{{< figure src="unregister_cmd_consumer.svg" title="Client unregisters the command consumer for a device" alt="A client sends a request message for removing the consumer registration entry and receives a response containing a confirmation" >}}
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a message to remove the consumer registration entry for a device in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name                  | Mandatory | Location                 | AMQP Type | Description |
+| :-------------------- | :-------: | :----------------------- | :-------- | :---------- |
+| *subject*             | yes       | *properties*             | *string*  | MUST be set to `unregister-cmd-consumer`. |
+| *adapter_instance_id* | yes       | *application-properties* | *string*  | The identifier of the protocol adapter instance to remove the registration entry for. Only if this adapter instance is currently associated with the device or gateway identified by the *device_id* property, the registration entry will be removed. |
+
+The body of the message SHOULD be empty and will be ignored if it is not.
+
+**Response Message Format**
+
+A response to a *unregister command consumer for device* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *204* | OK, the command consumer registration entry for the device has been removed. |
+| *400* | Bad Request, the request message does not contain all required properties. |
+| *412* | Precondition failed, the registration entry for the device has not been removed because there is no command consumer registration entry matching the adapter instance assigned to the device. |
```

Comment:
may we can add an example here for a sequence of actions that might lead to 412 being returned, e.g. a device losing its connection and re-connecting to a different adapter instance?

Reply:
Yes, I've added:
> This may happen if the mapping entry has already been removed or its lifespan has elapsed. The entry could also have been updated with a different adapter instance identifier before, which could for example mean the device lost its connection to the protocol adapter and has reconnected to a different adapter instance. The original adapter instance trying to remove the mapping entry at some point after that will result in a *412* error response because the instance identifier doesn't match.